ResistanceInAWire/resistance_in_a_wire_algorithm.py

```diff
@@ -0,0 +1,61 @@
+""" This contains an algorithm module that find the resistance in a wire
+ using functional programming. """
+
+# Define a lamba function to calculate the resistance of a wire using resitivity as rho,
+# length as l and cross-section area of wire as A.
+
+calculate_resistance = lambda rho, length, area: rho * length / area
+
+# Check during geneerator if user input invalid values.
+
+def resistance_generator(rho, area, lengths):
+    """Generator function that creates resistance values of a wire of different lengths"""
+
+    if area <= 0:
+        raise ValueError("The area entered must be above zero.")
+
+    if not lengths:
+        raise ValueError("You must input some value for the length.")
+
+    for length in lengths:
+        yield calculate_resistance(rho, length, area)
+
+# Adding user inputs for values of rho, length, and area.
+
+if __name__ == "__main__":
+    try:
+        # Asking user above for input values.
+        # Asking for resistivity.
+
+        rho = float(input(" Please enter the resistivity of your wire in ohm meters.: "))
```

Comment:
ResistanceInAWire/resistance_in_a_wire_algorithm.py:11:25: W0621: Redefining name 'rho' from outer scope (line 30) (redefined-outer-name)

You shouldn’t reuse variable names in different function in particular if they are the same. While this sounds counterintuitive at first and of course works in this case it can lead to problems once code is changed and you forget that this is not actually the same variable. So a better naming would be e.g.:
```python
rho_input = float(input(…))
```